packages/codecs-strings/src/utf8.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { Codec, combineCodec, Decoder, Encoder } from '@solana/codecs-core';
+
+import { removeNullCharacters } from './null-characters';
+
+/** Encodes UTF-8 strings using the native `TextEncoder` API. */
+export const getUtf8Encoder = (): Encoder<string> => {
+    let textEncoder: TextEncoder;
+    return {
+        description: 'utf8',
+        fixedSize: null,
+        maxSize: null,
+        encode: (value: string) => new Uint8Array((textEncoder ||= new TextEncoder()).encode(value)),
```

Comment:
Maybe now is the right time to make a `text-encoding-impl` package just like `fetch-impl` and `ws-impl` so that you can build separate browser/node and react-native bundles. The React Native one would basically take on some polyfill as a dependency.

Be sure to search for `fetch-impl` because there are some build scripts that have to be updated.